test/storage/index_manager_test.cpp

```diff
@@ -0,0 +1,117 @@
+  cols.emplace_back("id", type::TypeId::INTEGER, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("name", type::TypeId::VARCHAR, 100, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  cols.emplace_back("address", type::TypeId::VARCHAR, 200, false, catalog::col_oid_t(catalog_->GetNextOid()));
+  catalog::Schema schema(cols);
+
+  // create table
+  auto table = table_handle.CreateTable(txn0, schema, "test_table");
+  auto table_oid = table_handle.NameToOid(txn0, "test_table");
+  auto table_entry = table_handle.GetTableEntry(txn0, "test_table");
+  EXPECT_NE(table_entry, nullptr);
+  std::string_view str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(0));
+  EXPECT_EQ(str, "public");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(1));
+  EXPECT_EQ(str, "test_table");
+
+  str = type::TransientValuePeeker::PeekVarChar(table_entry->GetColInRow(2));
+  EXPECT_EQ(str, "pg_default");
+
+  // Insert a few rows into the table
+  auto ptr = table_handle.GetTable(txn0, "test_table");
+  EXPECT_EQ(ptr, table);
+
+  for (int i = 0; i < 200; ++i) {
+    std::vector<type::TransientValue> row;
+    std::ostringstream stringStream;
+    stringStream << "name_" << i;
+    std::string name_string = stringStream.str();
+    stringStream << "address_" << i;
+    std::string address_string = stringStream.str();
+    row.emplace_back(type::TransientValueFactory::GetBoolean(i % 2 == 0));
+    row.emplace_back(type::TransientValueFactory::GetInteger(i));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(name_string));
+    row.emplace_back(type::TransientValueFactory::GetVarChar(address_string));
+    ptr->InsertRow(txn0, row);
+  }
+
+  // Commit the setting transaction
+  txn_manager_->Commit(txn0, TestCallbacks::EmptyCallback, nullptr);
+
+  // Set up index attributes and key attributes
+  std::vector<std::string> index_attrs{"sex", "id", "name"};
+  std::vector<std::string> key_attrs{"id", "name"};
+
+  // Create the index
+  auto index_oid = index_manager_->CreateConcurrently(terrier_oid, ns_oid, table_oid, parser::IndexType::BWTREE, false,
+                                                      "test_index", index_attrs, key_attrs, txn_manager_, catalog_);
+
+  // Test whether the catalog has the corresponding information
+  auto txn1 = txn_manager_->BeginTransaction();
+  EXPECT_GT(!index_oid, 0);
+  auto index_handle = db_handle.GetIndexHandle(txn1, terrier_oid);
+  auto index_entry = index_handle.GetIndexEntry(txn1, index_oid);
+  EXPECT_NE(index_entry, nullptr);
+  auto ret = index_entry->GetIntegerColumn("indexrelid");
+  EXPECT_EQ(ret, !index_oid);
+  EXPECT_EQ(index_entry->GetIntegerColumn("indrelid"), !table_oid);
+  EXPECT_EQ(index_entry->GetBooleanColumn("indisready"), false);
+  EXPECT_EQ(index_entry->GetBooleanColumn("indisvalid"), true);
+  txn_manager_->Commit(txn1, TestCallbacks::EmptyCallback, nullptr);
```

Comment:
There is no check on whether the index itself is correct. As said above, I think a better test case is to start some active txns, then run `CreateConcurrently` and check if it builds the correct index without blocking the active txns. The current test case does not show the feature of `CreateConcurrently`.

Reply:
Fixed! Thank you.